sycl/test-e2e/VirtualMem/memory_granularity.cpp

```diff
@@ -0,0 +1,76 @@
+// REQUIRES: aspect-ext_oneapi_virtual_mem
+
+// This test checks whether memory granularity returned is greater then 0 and
+// that the recommended granularity is the same size or greater than the minimum
+// granularity for virtul memory extension.
+
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+#include <string_view>
+
+#include <sycl/detail/core.hpp>
+
+#include <sycl/ext/oneapi/virtual_mem/virtual_mem.hpp>
+
+namespace syclext = sycl::ext::oneapi::experimental;
+
+template <typename... ArgsT>
+int CheckReturnedGranularitySize(std::string_view TestName, ArgsT... Args) {
+  if (syclext::get_mem_granularity(Args...) == 0) {
+    std::cout << "Failed check: " << TestName << std::endl;
+    return 1;
+  }
+  return 0;
+}
+
+int main() {
+  int Failed = 0;
+  sycl::queue Q;
+  sycl::context Context = Q.get_context();
+  sycl::device Device = Q.get_device();
+
+  Failed += CheckReturnedGranularitySize(
+      "returned granularity size must be greater then 0 for device and "
+      "granularity_mode::recommended",
```

Comment:
```suggestion
      "recommended granularity size must be greater then 0 for device",
```

I think that we can simplify those messages like this, it should be mor or less obvious that we are talking about the value that is returned from some SYCL API